### PR TITLE
Format libcudacxx/include files without extensions

### PR DESCRIPTION
**Pull request by bdice**
## Description

This PR extends #1670 by adding rules to `pre-commit` that invoke `clang-format` over all extensionless headers in libcudacxx.

## Checklist
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: libcudacxx/include/cuda/std/type_traits [hunk 1]
```diff
@@ -13,10 +13,8 @@
 
 #include <cuda/std/detail/__config>
 
+#include <cuda/std/detail/__pragma_pop>
 #include <cuda/std/detail/__pragma_push>
-
 #include <cuda/std/detail/libcxx/include/type_traits>
```
[the remark this answers is not in the document]
Comment (bdice) [reply libcudacxx/include/cuda/std/type_traits:19]: @miscco and I discussed this offline. He is going to push changes to this PR that disable clang-format in these locations.

[the remark this answers is not in the document]
Comment (miscco) [reply libcudacxx/include/cuda/std/type_traits]: Done

**Review by miscco — APPROVED** [review]

**Comment by miscco** [comment]
Thanks a  lot for helping fixing this

**Pull request merged**